tests/src/integration_test.rs

```diff
@@ -0,0 +1,110 @@
+/* Copyright 2024 Canonical Ltd.
+ * All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Written by:
+ *        Nadzeya Hutsko <nadzeya.hutsko@canonical.com>
+ */
+
+use anyhow::Result;
+use std::path::PathBuf;
+use simple_test_case::test_case;
+use hwlib::{
+    models::request_validators::{CertificationStatusRequest, Paths},
+    models::response_validators::CertificationStatusResponse,
+    send_certification_status_request,
+};
+
+fn get_test_device_paths(device_type: &str) -> Paths {
+    let base_path = PathBuf::from("/app/client/hwlib/test_data/amd64").join(device_type);
+    Paths {
+        smbios_entry_filepath: base_path.join("smbios_entry_point"),
+        smbios_table_filepath: base_path.join("DMI"),
+        cpuinfo_filepath: PathBuf::from("./none"),
+        max_cpu_frequency_filepath: base_path.join("cpuinfo_max_freq"),
+        device_tree_dirpath: PathBuf::from("./none"),
+        proc_version_filepath: base_path.join("version"),
+    }
+}
+
+#[test_case(
+    "dgx_station";
+    "dgx_station"  // test name
+)]
+#[test_case(
+    "dell_xps13";
+    "dell_xps13"
+)]
+#[test_case(
+    "thinkstation_p620";
+    "thinkstation_p620"
+)]
+#[tokio::test]
+async fn test_certification_request(dir_path: &str) -> Result<()> {
+    let api_url = std::env::var("API_URL")
+        .unwrap_or_else(|_| String::from("http://hwapi-server-integration:8080"));
+    let cert_request = CertificationStatusRequest::new(get_test_device_paths(dir_path))?;
+    let response = send_certification_status_request(api_url, &cert_request).await?;
+
+    match response {
+        CertificationStatusResponse::Certified {
+            architecture,
+            bios: _,
+            board: _,
+            chassis: _,
+            available_releases,
```

Comment:
rather than underscoring the variables you don't need you can use `..` to discard them from the match:
```rust
CertificationStatusResponse::Certified {
    architecture,
    available_releases,
    ..
},
```